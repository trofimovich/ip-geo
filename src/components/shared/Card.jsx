import React from 'react';
import { string } from 'prop-types';

const Card = ({ title, content }) => (
  <div className="card">
    <div className="card-body">
      <h5 className="card-title">{title}</h5>
      <p className="card-text">{content}</p>
    </div>
  </div>
);

export default Card;

Card.defaultProps = {
  title: 'Title',
  content: 'content',
};

Card.propTypes = {
  title: string,
  content: string,
};
