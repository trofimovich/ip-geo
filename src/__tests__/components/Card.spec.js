import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Card from '../../components/shared/Card';

describe('Card', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <Card />,
    );
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Renders correctly with params', () => {
    const wrapper = shallow(
      <Card
        title="ABCD"
        contend="abcdefghijklmn"
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Renders loader when "isLoading" param is passed', () => {
    const wrapper = shallow(
      <Card
        isLoading
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
