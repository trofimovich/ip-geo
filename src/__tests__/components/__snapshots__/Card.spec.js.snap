// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Title
    </h5>
    <p
      className="card-text"
    >
      content
    </p>
  </div>
</div>
`;

exports[`Card Renders correctly with params 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      ABCD
    </h5>
    <p
      className="card-text"
    >
      content
    </p>
  </div>
</div>
`;

exports[`Card Renders loader when "isLoading" param is passed 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    />
  </div>
</div>
`;
